cpp/CMakeLists.txt

```diff
@@ -168,135 +168,335 @@ endif()
 #       which should give us a better parallel schedule.
 
 set(CUGRAPH_SOURCES
-    src/utilities/shuffle_vertices.cu
-    src/detail/permute_range.cu
-    src/utilities/shuffle_vertex_pairs.cu
-    src/detail/collect_local_vertex_values.cu
-    src/detail/groupby_and_count.cu
-    src/detail/collect_comm_wrapper.cu
-    src/sampling/random_walks_mg.cu
-    src/community/detail/common_methods_mg.cu
-    src/community/detail/common_methods_sg.cu
-    src/community/detail/refine_sg.cu
-    src/community/detail/refine_mg.cu
-    src/community/edge_triangle_count_sg.cu
-    src/community/edge_triangle_count_mg.cu
-    src/community/detail/maximal_independent_moves_sg.cu
-    src/community/detail/maximal_independent_moves_mg.cu
-    src/detail/utility_wrappers.cu
-    src/structure/graph_view_mg.cu
-    src/structure/remove_self_loops.cu
-    src/structure/remove_multi_edges.cu
-    src/utilities/path_retrieval.cu
+    src/utilities/shuffle_vertices_mg.cu
+    src/detail/permute_range_32.cu
+    src/detail/permute_range_64.cu
+    src/utilities/shuffle_vertex_pairs_mg.cu
```

Comment:
Don't we need to further split this file? This is one of the most problematic files.